### PR TITLE
PSC-11 configured multi-target system

### PR DESCRIPTION
**Pull request by BohdanDatskiv**
https://trello.com/c/V1FD4JXb

### CODE REVIEW

File: Uni/Enviroment.swift [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        case uni
+        case lnu
+        case nulp
+    }
+
+    private let configKey = "config_name"
+    private let baseURLKey = "base_url"
+
+    private init() {
+        guard let fileURL = Bundle.main.url(forResource: "AppConfig", withExtension: "plist") else {
```
Comment (Alex-Chaku) [inline Uni/Enviroment.swift]: Why there so many separate "guard" statements, join them into the more logical groups

Comment (Alex-Chaku) [reply Uni/Enviroment.swift]: There are hardcoded "AppConfig" names for file names, more them into constants

Comment (RomanSlyepko) [inline Uni/Enviroment.swift]: hardcoded String filenames

File: Uni/Enviroment.swift [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        guard let url = result["base_url"] as? String else {
+            fatalError()
+        }
+        self.baseURL = url
+        guard let configName = result["config_name"] as? String else {
+            fatalError()
+        }
+        switch configName {
+        case "uni":
+            self.config = .uni
```
Comment (Alex-Chaku) [inline Uni/Enviroment.swift]: Never do such switch/case if you can just make enum Config: String, and then use Config(rawValue: ...)

File: Uni/Enviroment.swift [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    private init() {
+        guard let fileURL = Bundle.main.url(forResource: "AppConfig", withExtension: "plist") else {
+            fatalError("no plist file")
+        }
+        guard let data = try? Data(contentsOf: fileURL) else {
+            fatalError("could not get data")
+        }
+        guard let result = try?
+            PropertyListSerialization.propertyList(from: data, options: [], format: nil) as? [String: Any] else {
```
Comment (RomanSlyepko) [inline Uni/Enviroment.swift]: use Codable for decoding .plist

File: Uni/Enviroment.swift [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            fatalError("could not serialize data")
+        }
+        guard let url = result["base_url"] as? String else {
+            fatalError()
+        }
+        self.baseURL = url
+        guard let configName = result["config_name"] as? String else {
+            fatalError()
+        }
+        switch configName {
```
Comment (RomanSlyepko) [inline Uni/Enviroment.swift]: redundant `switch` for checking `configName`

**Pull request merged**